components/dashboards-web-component/src/designer/components/WidgetOptionsConfiguration.jsx

```diff
@@ -17,28 +17,141 @@
+                                        value={options[i].type.possibleValues[j]}
+                                        primaryText={options[i].type.possibleValues[j]}
+                                    />);
+                            }
+                        }
+                        CurrentWidgetOptionsInputs.push(
+                            <div>
+                                <SelectField
+                                    floatingLabelText={options[i].title}
+                                    onChange={(event, key, payload) => {
+                                        this.handleWidgetOptionSelectFieldEvent(event, key, payload);
+                                    }}
+                                    value={value}
+                                    id={options[i].id}
+                                >
+                                    {items}
+                                </SelectField>
+                            </div>);
+                        break;
+                    case 'BOOLEAN':
+                        CurrentWidgetOptionsInputs.push(
+                            <div>
+                                <Checkbox
+                                    id={options[i].id}
+                                    label={options[i].title}
+                                    labelStyle={labelStyle}
+                                    onCheck={(event, isInputChecked) => {
+                                        this.handleWidgetOptionCheckBoxEvent(event, isInputChecked, options);
+                                    }}
+                                    checked={value}
+                                />
+                            </div>);
+                        break;
+                    default :
+                        break;
+                }
+            }
+        }
+
+        return CurrentWidgetOptionsInputs;
     }
 
     doesSelectedWidgetHasOptions() {
-        const contentOptions = _.get(this.props.selectedWidget.props, 'configs.options');
         const configurationOptions = _.get(this.props.selectedWidgetConfiguration, 'configs.options');
-        return (!_.isEmpty(contentOptions) && !_.isEmpty(configurationOptions));
+        return (!_.isEmpty(configurationOptions));
+    }
+
+    handleWidgetOptionSelectFieldEvent(event, key, payload) {
+        const optionId = event.target.parentElement.parentElement.parentElement.id;
+        this.handleWidgetOutput(optionId, payload);
+    }
+
+    handleWidgetOptionTextFieldEvent(event, newValue) {
+        const optionId = event.target.id;
+        this.handleWidgetOutput(optionId, newValue);
+    }
+
+    handleWidgetOptionCheckBoxEvent(event, isInputChecked) {
+        const optionId = event.target.id;
+        this.handleWidgetOutput(optionId, isInputChecked);
+    }
+
+    handleWidgetOutput(optionId, optionValue) {
+        const options = this.props.selectedWidget.props.configs.options;
+        options[optionId] = optionValue;
```

Comment:
If `optionValue` is `null` then we need to set a default value.

Reply:
How about we pass `defaultValue` to this method. `handleWidgetOutput(optionId, optionValue, defaultValue)` and in the event listener
```js
(event, newValue) => this.handleWidgetOutput(currentOption.id, newValue, currentOption.defaultValue)
```